test/template-tests/testcases/deploymentTemplate/IDs-Should-Be-Derived-From-ResourceIDs.test.ps1

```diff
@@ -1,16 +1,26 @@
-﻿param(
+﻿<#
+.Synopsis
+    Ensures that all IDs use the resourceID() function.
+.Description
+    Ensures that all IDs use the resourceID() function, or resolve to parameters or variables that use the ResourceID() function.
+.Example
+    Test-AzureRMTemplate -TemplatePath .\100-marketplace-sample\ -Test IDs-Should-Be-Derived-From-ResourceIDs
+.Example
+    .\IDs-Should-Be-Derived-From-ResourceIDs.test.ps1 -TemplateObject (Get-Content ..\..\..\..\100-marketplace-sample\azureDeploy.json | ConvertFrom-Json) 
+#>
+param(
+# The template object (the contents of azureDeploy.json, converted from JSON)
 [Parameter(Mandatory=$true,Position=0)]
-$MainTemplateObject
-)
+$TemplateObject)
 
 # First, find all objects with an ID property in the MainTemplate.
-$ids = $MainTemplateObject  | Find-JsonContent -Key id -Value * -Like 
+$ids = $TemplateObject  | Find-JsonContent -Key id -Value * -Like

```

Comment:
this should be for any property that endswith "id", e.g.
fooId
barId
id
etc

Reply:
Agreed.

I'll need to extent Find-JsonContent to do that and am trying to keep changes more discreet.  
Approve this one and I'll make an additional with the changes for Find-JsonContent.

Reply:
do we have a workitem to track that?